src/cache/index.ts

```diff
@@ -112,45 +110,24 @@ export const calculateFeedId = <T extends EndpointGenerics>(
  *
  * @example
  * ```
- * calculateKey({ base: 'ETH', quote: 'BTC' }, ['base','quote'])
- * // equals `|base:eth|quote:btc`
+ * calculateKey({ base: 'ETH', quote: 'BTC' })
+ * // equals `{"base":"eth","quote":"btc"}`
  * ```
  */
 export const calculateKey = <CustomSettings extends SettingsMap>(
   data: unknown,
-  paramNames: string[],
   adapterConfig: AdapterConfig<CustomSettings>,
 ): string => {
   if (data && typeof data !== 'object') {
     throw new Error('Data to calculate cache key should be an object')
   }
 
-  const params = data as Record<string, unknown>
-
-  let cacheKey = ''
-  for (const paramName of paramNames) {
-    const param = params[paramName]
-    if (param === undefined) {
-      continue
-    }
-
-    cacheKey += `|${paramName}:`
-    switch (typeof param) {
-      case 'string':
-        cacheKey += param.toLowerCase()
-        break
-      case 'number':
-      case 'boolean':
-        cacheKey += param.toString()
-        break
-      case 'object':
-        // Force cache keys to only use performant properties of the input params.
-        // If the object were to be used, we'd have to sort its properties.
-        logger.debug(
-          `Property "${paramName}" in request parameters is of type object, and won't be used in the cacheKey`,
-        )
+  let cacheKey = JSON.stringify(data, (_, value) => {
+    if (value && typeof value === 'string') {
+      return value.toLowerCase()
     }
-  }
+    return value
+  })
```

Comment:
Does this mean that these inputs are now giving different cache keys?

A:

```
{
    "id": "1",
    "data": {
        "from": "ETH",
        "to": "USD",
        "overrides": {
            "coingecko": { "ETH": "ethereum"}
        }
    }
}
```

B:

```
{
    "id": "1",
    "data": {
        "from": "ETH",
        "to": "USD",
        "overrides": {
            "coingecko": { "ETH": "ethereum"},
            "coinpaprika": { "ETH": "ethereum"}
        }
    }
}
```

Reply:
If so, we need to block this PR and sit down and figure out how that is handled

Reply:
If I understand correctly, I think both of these reduce down to something like these. I tested with an adapter to be sure.
`{"base":"eth","quote":"usd"} {"base":"ethereum","quote":"usd"}`

The input is rebuilt in the validator based on the input params so it would only contain those fields. Although if input params are array or object types, we could potentially generate different keys for the same input. But when I talked to Alejo about it, he wasn't concerned.

Reply:
The key reasoning as to why this would be safe lies in where this is being used: we will always generate the cache key after the data has been validated. Therefore the only thing that should be present for the stringify call should be valid input parameters, and that's a known, reduced subset without meta-parameters like overrides. It's true, however, that this is not accounting for unsorted arrays or nested structures, that's something easy to tackle but should be done carefully to make sure it's as performant as possible (i.e., if possible avoid sorting at all costs)

Reply:
Ok cool, since it's post-validator then it alleviates my original concern